torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -63,6 +63,27 @@ def _init_mp_dtypes(self) -> None:
         self._orig_dtype = next(iter(orig_dtypes))
         self._param_dtype = self._orig_dtype
 
+    def _init_grad_divide_factors(self):
+        """
+        For N data parallel workers, each worker computes g_i, and they
+        collectively reduce to compute (g_1 + ... + g_N) / N. To avoid overflow
+        and underflow, we divide by sqrt(N) before and after the reduction.
+        """
+        data_parallel_world_size = 1
```

Comment:
why do you init this to 1 and multiply out?

also the comment says sqrt(N) but aren't you doing something that is loosely inspired by that but not equivalent?

Reply:
Looking at https://github.com/facebookresearch/fairscale/pull/565, I think this 'init to 1 and multiply out' is just a nice way of making sure the pre- and post-factors multiply to the world size while keeping them as integer values. I did not see any motivation for why they need to be integer though.